app/controllers/application_controller.rb

```diff
@@ -4,4 +4,9 @@ class ApplicationController < ActionController::Base
   def current_user
     @_current_user ||= UserFacade.search_user(session[:user_id]) if session[:user_id]
   end
+
+  def require_current_user
```

Comment:
I think we might need to check the logic here. The way I'm reading it the flash will send in both cases, but I could be wrong

Reply:
That looks like what I would expect